tests/test_hra.py

```diff
@@ -982,6 +911,147 @@ def test_exception_invalid_decay(self):
+            pygeoprocessing.numpy_array_to_raster(
+                array, nodata, (10, -10), ORIGIN, SRS_WKT,
+                pairwise_raster_dicts[0][key])
+
+        # For the sake of testing this function more rigorously, creating a new
+        # classification path for the per-habitat summary classification
+        # raster.
+        per_habitat_classifications = {
+            pairwise_raster_dicts[0]['habitat']: os.path.join(
+                self.workspace_dir, 'summary_classes.tif')
+        }
+        pygeoprocessing.numpy_array_to_raster(
+            numpy.array([[2, 3, 2, 3]], dtype=numpy.int8),
+            nodata, (10, -10), ORIGIN, SRS_WKT,
+            list(per_habitat_classifications.values())[0])
+
+        target_summary_csv_path = os.path.join(
+            self.workspace_dir, 'summary.csv')
+        aoi_vector_path = os.path.join(self.workspace_dir, 'aoi.shp')
+        subregion_bounding_box = pygeoprocessing.get_raster_info(
+            list(per_habitat_classifications.values())[0])['bounding_box']
+        subregion_geom = shapely.geometry.box(*subregion_bounding_box)
+
+        # This is a standard record in the summary table, used in both subtests
+        # below.
+        std_record = {
+            'HABITAT': pairwise_raster_dicts[0]['habitat'],
+            'STRESSOR': pairwise_raster_dicts[0]['stressor'],
+            'E_MIN': numpy.min(e_array),
+            'E_MAX': numpy.max(e_array),
+            'E_MEAN': numpy.sum(e_array) / 4,
+            'C_MIN': numpy.min(c_array),
+            'C_MAX': numpy.max(c_array),
+            'C_MEAN': numpy.sum(c_array) / 4,
+            'R_MIN': numpy.min(risk_array),
+            'R_MAX': numpy.max(risk_array),
+            'R_MEAN': numpy.sum(risk_array) / 4,
+            'R_%HIGH': 25.0,
+            'R_%MEDIUM': 25.0,
+            'R_%LOW': 25.0,
+            'R_%NONE': 25.0,
+        }
+
+        with self.subTest("multiple subregion names"):
+            # 3 subregions, 2 of which have the same name.
+            # In cases of overlap, the function double-counts.
+            pygeoprocessing.shapely_geometry_to_vector(
+                [subregion_geom] * 3, aoi_vector_path, SRS_WKT, 'ESRI Shapefile',
+                fields={'name': ogr.OFTString}, attribute_list=[
+                    {'name': 'first region'},
+                    {'name': 'first region'},
+                    {'name': 'second region'}
+                ])
+            hra._create_summary_statistics_file(
+                aoi_vector_path, pairwise_raster_dicts,
+                per_habitat_classifications, target_summary_csv_path)
+            expected_records = [
+                {**std_record,
+                 **{'SUBREGION': 'first region',
+                    'STRESSOR': '(FROM ALL STRESSORS)'},
+                    'R_%HIGH': 50.0,
+                    'R_%MEDIUM': 50.0,
+                    'R_%LOW': 0,
+                    'R_%NONE': 0,
```

Comment:
I'm a bit confused by these numbers. Somehow the % from all stressors are different than the % from one stressor, even though in this test there is only one stressor, right?

Reply:
Yes, if we were running the model with only one habitat/stressor pair, we would expect the cumulative risk to match the pairwise risk.  In this case in this test, I'm putting in different values for the cumulative risk classes to test the summary table construction only -- these values are definitely not real or reasonable model outputs!

I've added a note to clarify this (and improved variable names a bit) in 43a37ec52.

Reply:
Gotcha, thanks for adding that clarification in the test